### ISSUE TITLE
Make gEAR pages WCAG 2.1 Level AA compatible

### ISSUE THREAD

**Issue by adkinsrs** [IGS/gEAR#1220]
UMB recommends the Silktide Chrome extension to check this -> https://silktide.com/toolbar/

https://www.umaryland.edu/accessibility/

As part of Title II ADA federal requirements all university sites and instructional materials need to be WCAG compliant by April 26, 2026.  It is pretty tough for us to get everything perfect by then, but we should try to prioritize the most frequently accessed pages first.

**Comment by adkinsrs** [comment]
This should apply not only to our HTML but any HTML snippets created in our Javascript code.  I am unsure if our CSS in https://github.com/IGS/gEAR/tree/026856535c0f596d4fb3fd09e72a95860e1e2b27/www/css needs to be adjusted (the "bulma" and "gear-theme-purple" code would not be touched).

Another pain point that I may need to address separately is how best to handle our plotting results.  Plotly SVG output is not quite WCAG-ready, and for images created in Matplotlib, we need to figure out how to give the figure image an alt_text parameter that adequately describes the plot.  I think we can draw some inspiration from AltGosling (https://github.com/gosling-lang/altgosling) which could be used for Gosling displays.  We also need to consider the submitted SVG datasets, as well as spatial displays created in Panel.

We also need to finish colorblind accessibility options.  We currently have a user-level flag where the user can select if they need colorblind mode (visualizations change to a "cividis" palette) but there are a couple of locations where it still needs to be implemented.  One is in the spatial panel displays, one is for Gosling displays, and one is for all the plots in the single-cell workbench.

**Comment by adkinsrs** [comment]
Worth noting that we need to ensure future pages and additions must be compliant as well. It may be worthwhile to find a pre-commit hook that runs a compliance check and rejects a commit if the additions or changes do not match.

**Comment by jorvis** [comment]
Yep, there seem to be several options out there, including:

https://github.com/qed42/ai-accessibility-checker

**Comment by adkinsrs** [comment]
For "descriptive" tasks, I mean add info explaining what plot type is used against what datasets and what conditions.  Not to make an attempt to interpret the data.

- [x] Test initial Copilot edits to pages (+ use Silktide extension to verify compliance)
- [ ] Finish colorblindness viewing for all additional plots that need it.  
- [x] (LOWER PRIORITY) I also realized that the colorblindness check is bound by user settings, so any non-logged in user cannot access it. Not sure if this warrants an adjustment or not.
- [x] Add "aria" labels on SVG plot types
- [ ] Add descriptive title to the spatial panel "iframe" tag
- [x] Add descriptive alt_text to scanpy images
- [ ] Add descriptive "aria" label to Plotly figures
- [ ] (LOWER PRIORITY) Add a "description" tag to dataset displays, and when saving a curation in one of the UIs, enforce a requirement that description needs to be filled.  This would allow us to use it in the alt-text and aria labels directly.  Would need backup methods for legacy datasets (not all users would add labels to their saved curations or add meaningless names).

**Comment by adkinsrs** [comment]
Posting some notes with the university team I met with last week

* The Title II compliance applies more to the public and students of your university (i.e. coursework).  Since gEAR is accessed mainly by domain-specific researchers, it is considered lower-risk
* Title II also gives leeway for “good faith” implementation and doing your best. To me combined with our target audience, that gives us probably some leeway to make changes at a slower pace.
* I showcased what I had done so far with respect to visualizations and alt-text.  They think that was a good start and feel that maybe instead of trying to describe the plot, it is fine to describe the dataset.  
  * In which case, we need to make it easier to access the dataset descriptions from the “gene expression view”.  If any prior datasets do not have a dataset description, attempt to reach out and backfill those, or maybe get from pubmed.  For future datasets, require the description.

* We have like 60 tabbable elements on the “gene expression view” page, with dataset dropdowns menus way in the back.  Perhaps we should refine the tab priority after the view renders.
* Talked about the hover annotations for plotly, and how they may not be accessible.  They cited the relatively new SVG hovers I implemented as a better approach.
* https://gosling-lang.github.io/altgosling/. I showed AltGosling as what I feel is a good accessibility plotting implementation, and they liked it.  I did note the engineering effort to make this for all plot types we have. One of the people suggested that perhaps, we have a button that opens a new tab with this view.
  * After the meeting another attendee posted a couple of resources, and while they are centric to the the tool, I think we could use some Tableau documentation for inspiration
    * [Tips for Content Authors](https://help.tableau.com/current/pro/desktop/en-us/accessibility_create_view.htm)
    * [Designing Accessible Views](https://help.tableau.com/current/pro/desktop/en-us/accessibility_best_practice.htm)
* I showed the colorblindness mode in our user settings.  I kind of knew this would be asked, but they wondered if it was possible to implement the setting as maybe a button on the page so that the ability is not bound by having a gEAR account.
  * What I did to address this is to add an accessibility widget button to all gEAR pages as seen in commit 4fd29445d5f4ec74248332db125a09cac3b852d1

**Comment by adkinsrs** [comment]
Any future work will be in new tickets
